### DISCUSSION TITLE
Manually call text highlighting with text to highlight

### DISCUSSION THREAD

**Discussion by DFournierIT** [wet-boew/wet-boew#9603] [Q&A] [answered]
Can text highlighting be called manually while specifying the text to highlight? If so, what would that look like?

Essentially, I'm doing a POST and would need to call the text highlighting depending on what is sent with the form. At this point, a GET isn't feasible due to the huge amount of possible parameters that could overwhelm the querystring and cause an HTTP 414 (URI Too Long).

**Comment by wewhite** [comment, marked as answer]
Yes it can be called manually, since v4.0.7
See this:  Texthighlight plugin #6126 
And this PR:  Text Highlight: added support for arrays and added more test cases #6137 
This comment states it still needs to be documented: https://github.com/wet-boew/wet-boew/issues/6126#issuecomment-58215530

Basically, you can do this in JavaScript, calling the plugin manually, looks like this:
```

    // get the pre-defined area to search and highlight
    const elm = document.getElementById("searchAbleArea");

    // define strings to search for
    var $searchCriteria1 = 'style'; // single string
    var $searchCriteria2 = ['style','words', 'to', 'be', 'highlighted']; // string array
    var $searchCriteria3 = 'style|words|to|be|highlighted'; // this or that

    // wait for all WET to finish 
    $( document ).on( "wb-ready.wb", function( event ) {

        // add the Text Highlighting Plugin class name to the pre-defined area
        elm.classList.add("wb-txthl");

        // manually trigger the plugin type, and extra paramters named txthl 
        $( ".wb-txthl" ).trigger( {type: 'wb-init.wb-txthl', txthl: $searchCriteria3} );
      
    });
    
    // Text Highlighting plugin finished initializing
    $( ".wb-txthl" ).on( "wb-ready.wb-txthl", function( event ) {
        // ...
    });
```

Key point is to manually trigger the plugin, passing it as an Object, including the `type: 'wb-init.wb-txthl'` and `txthl: ...` parameter.

**Reply by DFournierIT** [reply]
Thanks, this worked!
